connectors/src/connectors/interface.ts

```diff
@@ -12,96 +10,72 @@ import type { ConnectorConfiguration } from "@dust-tt/types";
 
 import type { DataSourceConfig } from "@connectors/types/data_source_config";
 
-export type ConnectorCreate<T extends ConnectorConfiguration> = (
-  DataSourceConfig: DataSourceConfig,
-  connectionid: string,
-  configuration: T
-) => Promise<Result<string, Error>>;
+export abstract class BaseConnectorManager<T extends ConnectorConfiguration> {
```

Comment:
I think this class is a solid foundation, but it could include more functionality to prevent redundant implementations across all connectors. For instance, if the child classes implement `sync` and `stop`, we could potentially move the `clean`/`pause`/`unpause` logic into this class, allowing each connector to override it if necessary. WDYT?

Reply:
I don't see how we'd do that ? 🤔 

Anyway in this PR I want to keep the behaviour / logic change to a minimum. Happy to iterate in subsequent PRs 👍